tests/test_create_pod_attaching_multiple_pvc_start_IO.py

```diff
@@ -0,0 +1,376 @@
+        yaml_data
+    )
+    cirros_pod_name = data['metadata']['name']
+    temp_dict1 = []
+    temp_dict2 = []
+    temp_str = ''
+    temp_list1 = data['spec']['template']['spec']['containers'][0]['args']
+    temp_list1.pop(len(temp_list1) - 1)
+    temp_list2 = data['spec']['template']['spec']['containers'][0][
+        'livenessProbe']['exec']['command']
+    temp_list2.pop(len(temp_list2) - 1)
+    for i in pv_list:
+        temp_dict1.append(
+            {'name': i, 'persistentVolumeClaim': {'claimName': i}}
+        )
+        temp_dict2.append({'mountPath': '/mnt' + i, 'name': i})
+        temp_list2.append(
+            'mount | grep /mnt' + i + ' && head -c 1024 '
+            '< /dev/urandom >> /mnt' + i + '/random-data.log'
+        )
+        temp_str = temp_str + '(mount | grep /mnt' + i + ') && (head -c ' \
+            '1000 < /dev/urandom > /mnt' + i + '/random-data.log) || exit 1;'
+    del data['spec']['template']['spec']['volumes']
+    del data['spec']['template']['spec']['containers'][0]['volumeMounts']
+    del data['spec']['template']['spec']['containers'][0][
+        'livenessProbe']['exec']['command']
+    data['spec']['template']['spec']['volumes'] = temp_dict1
+    data['spec']['template']['spec']['containers'][0][
+        'volumeMounts'] = temp_dict2
+    temp_list1.append('while true; do ' + temp_str + ' sleep 20 ; done')
+    data['spec']['template']['spec']['containers'][0][
+        'livenessProbe']['exec']['command'] = temp_list2
+    with open(TEMP_YAML, 'w') as yaml_data:
+        yaml.dump(data, yaml_data, default_flow_style=False)
+    log.info(f"Create cirros pod with all the created pvcs")
+    assert DC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    try:
+        # Check for pod is up and running
+        POD.wait_for_resource(
+            condition='Running', selector='deploymentconfig=cirrospod',
+            timeout=180, sleep=20
+        )
+        log.info(f"Pod {cirros_pod_name} creation success")
+        return cirros_pod_name
+    except exceptions.CommandFailed:
+        log.info(f"Exception pod {cirros_pod_name} is not in Running state")
+        return False
+
+
+def delete_pod(pod_name):
+    """
+    Delete the cirros pod
+
+    """
+    try:
+        cmd = f"delete deploymentconfig {pod_name}"
+        assert DC.exec_oc_cmd(command=cmd)
+        # Wait for pod deletion to complete, else pvc remove will fail
+        sleep(40)
+        return True
+    except exceptions.CommandFailed:
+        log.info(f'Exception POD {pod_name} is not terminated ')
+        return False
+
+
+def delete_persistent_volume(pv_list):
+    """
+    Delete a Persistent Volume by given name
+    """
+    for i in pv_list:
+        try:
+            cmd = f"delete pvc {i}"
+            assert PVC.exec_oc_cmd(command=cmd)
+        except exceptions.CommandFailed:
+            log.info(f'Exception PVC {i} is not deleted')
+            return False
```

Comment:
I would do something bit different.
let's say you have 100 pvcs.
the 10th failed to delete so you will end up with 91 pvcs as leftovers.
I would have a list, and every pvc that failed to delete append to the list and in case you have something in that list after tried to delete all pvcs, return False

Reply:
Good suggestion, I will implement it.